src/todo_file/src/lib.rs

```diff
@@ -201,6 +205,12 @@ impl TodoFile {
 		self.history.reset();
 	}
 
+	/// Set the rebase todo file state.
+	#[inline]
+	pub fn set_state(&mut self, state: State) {
+		self.state = state;
```

Comment:
We will want/need to track this in `History` (`self.history`), so that undo/redo respects the state change.

The complication here, is that `state` is a side effect, so it will need to be handled a little differently than the existing history. 🤔